pkg/webui/console/store/middleware/logics/applications.js

```diff
@@ -25,9 +25,11 @@ import createEventsConnectLogics from './events'
 
 const getApplicationLogic = createRequestLogic({
   type: applications.GET_APP,
-  async process ({ action }) {
+  async process ({ action }, dispatch) {
     const { payload: { id }, meta: { selector }} = action
-    return api.application.get(id, selector)
+    const app = await api.application.get(id, selector)
```

Comment:
This doesn't seem right to me; adding the dispatch here mixes responsibilities. Can we not solve this by orchestrating the dispatches in the component (e.g. using the promise and async/await)?

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/commit/ebdc6b422eceda37a30b06c850d8f00fca65ebbd#diff-b88e01b98363d1c7259c98f3dd34cf7b